src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py

```diff
@@ -243,15 +243,15 @@ def enable_model_cpu_offload(self, gpu_id=0):
             self.to("cpu", silence_dtype_warnings=True)
             torch.cuda.empty_cache()  # otherwise we don't see the memory savings (but they probably exist)
 
-        hook = None
-        for cpu_offloaded_model in [self.text_encoder, self.unet, self.vae]:
-            _, hook = cpu_offload_with_hook(cpu_offloaded_model, device, prev_module_hook=hook)
+            hook = None
+            for cpu_offloaded_model in [self.text_encoder, self.unet, self.vae]:
+                _, hook = cpu_offload_with_hook(cpu_offloaded_model, device, prev_module_hook=hook)
 
-        if self.safety_checker is not None:
-            _, hook = cpu_offload_with_hook(self.safety_checker, device, prev_module_hook=hook)
+            if self.safety_checker is not None:
+                _, hook = cpu_offload_with_hook(self.safety_checker, device, prev_module_hook=hook)
 
-        # We'll offload the last model manually.
-        self.final_offload_hook = hook
+            # We'll offload the last model manually.
+            self.final_offload_hook = hook
```

Comment:
hmm, I think it might be cleaner to just initially check if the hooks are already installed and then early return?